kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/ReadMetadata.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.benchmarks;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.nio.file.Path;
+
+/** Workload specification for read_metadata benchmarks. */
```

Comment:
I also wonder -- would we ever want a ReadMetadataSpec (for some table, at some version) to be extended to also be a ReadDataSpec?

i.e. would that class extend this class? like ... is a workload spec in JSON really just "read metadata" or is it really just "read" and then different runners can test reading metadata or reading data, too ?

Reply:
Hmm that's an interesting point. I'd  initially intended for readMetadata and readData to be separate workload spec and separate runners. 

This property actually becomes very handy with how the polymorphism works. It all becomes a simple flow like this:
```
WorkloadSpec spec = objectMapper.read(json_file)
WorkloadRunner runner = spec.getRunner(...)
runner.executeAsBenchmark()
```

We could introduce a flag in the WorkloadSpec itself. That way, getRunner(..) constructs the right one.


I think introducing this shared notion at the spec is interesting. Given that we're not implementing the data phase just yet, I'd advocate to punt on this design for now. Renaming is easy since this is private after all :)